<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ScriptAppPage</class>
 <widget class="QWidget" name="ScriptAppPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>512</width>
    <height>290</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Script</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QRadioButton" name="executableRadio">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KUrlRequester" name="executablePath">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="clickMessage">
           <string>Enter the absolute path to an script file</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QRadioButton" name="projectTargetRadio">
          <property name="text">
           <string/>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="ProjectItemLineEdit" name="projectTarget">
          <property name="clickMessage">
           <string>Enter a project target as path in the tree</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Script:</string>
        </property>
        <property name="buddy">
         <cstring>executableRadio</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Project Target:</string>
        </property>
        <property name="buddy">
         <cstring>projectTargetRadio</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Interpreter:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="KLineEdit" name="interpreter"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Behaviour</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Arguments:</string>
        </property>
        <property name="buddy">
         <cstring>arguments</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="KLineEdit" name="arguments">
        <property name="toolTip">
         <string>Enter arguments to give to the executable</string>
        </property>
        <property name="clickMessage">
         <string>Enter arguments to give to the executable</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Working Directory:</string>
        </property>
        <property name="buddy">
         <cstring>workingDirectory</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="KUrlRequester" name="workingDirectory">
        <property name="toolTip">
         <string>Select a working directory for the executable</string>
        </property>
        <property name="clickMessage">
         <string>Select a working directory for the executable</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Environment:</string>
        </property>
        <property name="buddy">
         <cstring>environment</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="KDevelop::EnvironmentSelectionWidget" name="environment">
        <property name="toolTip">
         <string>Select an environment to be used</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>External Terminal:</string>
        </property>
        <property name="buddy">
         <cstring>runInTerminal</cstring>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="runInTerminal">
        <property name="toolTip">
         <string>When enabled, runs the application in an external terminal.</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QFrame</extends>
   <header>kurlrequester.h</header>
  </customwidget>
  <customwidget>
   <class>KLineEdit</class>
   <extends>QLineEdit</extends>
   <header>klineedit.h</header>
  </customwidget>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KDevelop::EnvironmentSelectionWidget</class>
   <extends>KComboBox</extends>
   <header>util/environmentselectionwidget.h</header>
  </customwidget>
  <customwidget>
   <class>ProjectItemLineEdit</class>
   <extends>KLineEdit</extends>
   <header>util/projectitemlineedit.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>projectTargetRadio</tabstop>
  <tabstop>projectTarget</tabstop>
  <tabstop>executableRadio</tabstop>
  <tabstop>executablePath</tabstop>
  <tabstop>arguments</tabstop>
  <tabstop>workingDirectory</tabstop>
  <tabstop>environment</tabstop>
  <tabstop>runInTerminal</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>projectTargetRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>projectTarget</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>142</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>337</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>projectTargetRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>executablePath</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>142</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>384</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>executableRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>projectTarget</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>142</x>
     <y>87</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>executableRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>executablePath</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>142</x>
     <y>87</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
